src/libcrun/cgroup.c

```diff
@@ -260,7 +260,8 @@ int
 libcrun_cgroup_enter (struct libcrun_cgroup_args *args, struct libcrun_cgroup_status **out, libcrun_error_t *err)
 {
   __attribute__ ((unused)) pid_t sigcont_cleanup __attribute__ ((cleanup (cleanup_sig_contp))) = -1;
-  cleanup_cgroup_status struct libcrun_cgroup_status *status;
+  /* status will be filled by the cgroup manager.  */
+  cleanup_cgroup_status struct libcrun_cgroup_status *status = xmalloc0 (sizeof *status);
```

Comment:
LGTM

However we allocate early even if we are not going to use this variable. But we have `cleanup` so should be good.